RFS/src/main/java/com/rfs/worker/MetadataStep.java

```diff
@@ -32,6 +33,16 @@ public SharedMembers(GlobalState globalState, CmsClient cmsClient, String snapsh
             this.metadataFactory = metadataFactory;
             this.metadataCreator = metadataCreator;
             this.transformer = transformer;
+            this.cmsEntry = null;
+        }
+
+        // A convient way to check if the CMS entry is null before retrieving it.  In some places, it's fine/expected
+        // for the CMS entry to be null, but in others, it's a problem.
+        public CmsEntry.Metadata getCmsEntryNotNull() {
+            if (cmsEntry == null) {
+                throw new MissingMigrationEntry("The Metadata Migration CMS entry we expected to be stored in local memory was null");
+            }
+            return cmsEntry;
```

Comment:
Look into `Optional<CmsEntry.Metadata>` as it includes `getOrThrow()` as a way to cleanly handle or bubble up exceptions if this design pattern is needed

Reply:
Switched to `Optional`'s, but kept this method as it makes the places it's called a bit less busy